### PR TITLE
feat: node versioning (WIP)

### PR DESCRIPTION
**Pull request by rarkins**
Adds "nodeVer" versioning, which follows semver so far except that it makes all non-supported and non-LTS versions of node as "unstable". This results in behaviour like:
- Upgrades < 8 will be upgraded to 8.x and not further
- Anyone on 9.x will be upgraded to 10.x

To be decided:
- Does it cause any confusion or disagreement that we are using "unstable" for all releases < 6? Technically 4.x is deprecated not unstable, but right now it doesn't make any difference to the end result.
- What extra work to link this with `supportPolicy` (LTS, etc). Would anything in `supportPolicy` contradict this behaviour?
- Anything extra required for Travis to work with this?

### CODE REVIEW

**Comment by rarkins** [comment]
Cc @destroyerofbuilds @olstenlarck

**Comment by olstenlarck** [comment]
I'm not very familiar with that "versioning" thing, what's that `nodeVer`? Does that mean, that if we just add `supportPolicy: lts_active` we should expect only bumps for the 6 and 8 versions currently, and later 8 and 10? 

> Does it cause any confusion or disagreement that we are using "unstable" for all releases < 6? 

No, i think.

**Comment by destroyerofbuilds** [comment]
> Does it cause any confusion or disagreement that we are using "unstable" for all releases < 6? Technically 4.x is deprecated not unstable, but right now it doesn't make any difference to the end result.

Yes, I think `unstable` is the wrong word to use. As you noted, 4.x is _deprecated_. It could also be called _unsupported_.

File: lib/versioning/nodever/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const semver = require('../semver');
+
+// Only LTS is stable
+const isStable = version =>
+  semver.isStable(version) && semver.matches(version, '^6.9.0 || ^8.9.0');
```
Comment (destroyerofbuilds) [inline lib/versioning/nodever/index.js]: Are we still waiting on https://github.com/nodejs/Release/issues/290 so that these hard-coded values can go away?

Comment (rarkins) [reply lib/versioning/nodever/index.js]: Pretty much, yes. I think though that node.js plan far enough in advance so we could add conditional checks on dates too to remove the need for a commit/update on the exact days(s) that Node.js changes status.

**Comment by rarkins** [comment]
@olstenlarck wrote:
> Does that mean, that if we just add `supportPolicy: lts_active` we should expect only bumps for the 6 and 8 versions currently, and later 8 and 10?

Yes, that would be my intention. For Travis that is easy - it's an array and we put in all the values necessary. Are there any other common uses cases/package files where you will have *both* versions? Otherwise everywhere in your code would get upgraded to 8.x anyway, even without a supportPolicy.

**Comment by olstenlarck** [comment]
Cool! :)

> Are there any other common uses cases/package files where you will have both versions?

No, i don't think so.

**Comment by kayoub5** [comment]
@rarkins consider pulling data from https://github.com/nodejs/Release/blob/master/schedule.json instead of hardcoding LTS releases.

**Comment by iamstarkov** [comment]
there is also one source closer to actual distributives https://nodejs.org/dist/index.json

**Comment by iamstarkov** [comment]
I'm confused. @rarkins, what did you conclude?

**Comment by rarkins** [comment]
@iamstarkov I decided to abandon this PR as right now it’s pretty pointless, and discuss more in https://github.com/renovatebot/renovate/issues/2443

**Pull request closed without merging**